### PR TITLE
Fix generation code and add Github actions to generate and deploy report

### PR DESCRIPTION
**Pull request by lucas-koehler**
### Fix type errors and adapt to VSCode repository changes 

* Avoid undefined type errors in comparator code
* Fix VSCode URL pattern and branch name
  * VSCode's typescript definition files were moved to a separate folder
  * VSCode renamed its master branch to main
  * Handle current and pre 1.63 versions of VSCode

### Add Github actions workflow configurations 

* Builds a new version of the report on merge to master and nightly.
* Build report code on pull request (CI)

**Note:** The deployment might need additional configuration in the repository.

Contributed on behalf of STMicroelectronics

Signed-off-by: Lucas Koehler <lkoehler@eclipsesource.com>

### CODE REVIEW

**Comment by JonasHelming** [comment]
fixes #6

**Review by JonasHelming — APPROVED** [review]

**Comment by JonasHelming** [comment]
The repository requires commit signing (PGP), I will open a bug to web master to remove this.

**Review by JonasHelming — APPROVED** [review]

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Generate and publish report nightly
+on:
+  schedule:
```
Comment (benoitf) [inline .github/workflows/nightly.yml:3]: why not add this cron scheduler to the previous job ?

Comment (lucas-koehler) [reply .github/workflows/nightly.yml]: Good catch, thank you :)
I merged nightly.yml and merge.yml together and renamed it to `report.yml` to better reflect the contents. I amended the previous commit.

**Review by benoitf — APPROVED** [review]

**Pull request merged**